cpmpy/transformations/linearize.py

```diff
@@ -572,3 +623,38 @@ def only_positive_coefficients(lst_of_expr):
 
     return newlist
 
+def linearize_objective(expr, supported=frozenset(["sum","wsum"])):
+    """
+    Takes a linear expression and outputs:
+    
+    - Decision variable: Var
+    - Linear: sum([Var])                                   (CPMpy class 'Operator', name 'sum')
+              wsum([Const],[Var])                          (CPMpy class 'Operator', name 'wsum')
+
+    Only difference with flatten_objective is that no negated boolvars are allowed.
+    """
+    
+    flatexpr, flatcons = flatten_objective(expr, supported=supported)
+    
+    if isinstance(flatexpr, _NumVarImpl) or flatexpr.name in {"sum","wsum"}:
+        pos_expr, const = only_positive_bv_linear(flatexpr)
+        if (const != 0):
+            assert isinstance(pos_expr, Operator) and pos_expr.name == "wsum", f"unexpected expression, should be wsum but got {pos_expr}"
+            pos_expr = Operator("wsum", [pos_expr.args[0]+[1], pos_expr.args[1] + [const]])
+        return pos_expr, flatcons
+    
+    else: 
+        assert flatexpr.name in supported, (f"Unexpected numerical expression, {flatexpr} is not linear and not supported, please report on github")
+        # other operators in expression such as "min", "max"
+        new_cons = []
+        nbv_sel = [isinstance(a, NegBoolView) for a in flatexpr.args]
+        if any(nbv_sel):
+            posexpr = copy.copy(flatexpr)
+            for i,arg in enumerate(list(posexpr.args)):
```

Comment:
you shouldn't loop over something that you change inside; which you hide by taking a copy with list(), but you already have a copy... namely the flatexpr; plus you don't need to do the isinstance again, because you have it in nbv_sel...

```
if any(nbv_sel):
   newargs = []
   for i,arg in enumerate(flatexpr.args):
     if nbv_sel[i]:
        # same comment as above about making an intvar where a boolvar could do
        newargs.append(aux)
      else:
        newargs.append(arg)
      posexpr = copy
      posexpr.update_args(newargs)  # the only recommended way to update args of a fresh copy!
```